lib/improver/tests/nowcasting/optical_flow/test_generate_optical_flow_components.py

```diff
@@ -0,0 +1,103 @@
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for generation of optical flow components"""
+
+import unittest
+from datetime import datetime
+
+import iris
+import numpy as np
+from iris.tests import IrisTest
+
+from improver.nowcasting.optical_flow import generate_optical_flow_components
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+from improver.utilities.warnings_handler import ManageWarnings
+
+
+class Test_generate_optical_flow_components(IrisTest):
+    """Tests for the generate_optical_flow_components method"""
+
+    def setUp(self):
+        """Set up test cubes.  Optical flow velocity values are tested within
+        the Test_optical_flow module; this class tests timestamps only."""
+        self.iterations = 20
+        self.ofc_box_size = 10
+
+        dummy_cube = set_up_variable_cube(
+            np.zeros((30, 30), dtype=np.float32),
+            name="lwe_precipitation_rate", units="mm h-1",
+            spatial_grid="equalarea", time=datetime(2018, 2, 20, 4, 0),
+            frt=datetime(2018, 2, 20, 4, 0))
+        coord_points = 2000*np.arange(30, dtype=np.float32)  # in metres
+        dummy_cube.coord(axis='x').points = coord_points
+        dummy_cube.coord(axis='y').points = coord_points
+
+        self.first_cube = dummy_cube.copy()
+        self.second_cube = dummy_cube.copy()
+        # 15 minutes later, in seconds
+        self.second_cube.coord("time").points = (
+            self.second_cube.coord("time").points + 15*60)
+
+        self.third_cube = dummy_cube.copy()
+        # 30 minutes later, in seconds
+        self.third_cube.coord("time").points = (
+            self.third_cube.coord("time").points + 30*60)
+
+        self.expected_time = self.third_cube.coord("time").points[0]
+
+    @ManageWarnings(ignored_messages=[
```

Comment:
You could add an extra test for a cubelist with less cubes in. What happens if you only pass in two cubes? Or do you think this is something we don't need to worry about?

Reply:
Turns out, what happens is it doesn't work.  Good thing we moved this out of the CLI!  Will fix.